homeassistant/components/buienradar/camera.py

```diff
@@ -60,7 +60,13 @@ class BuienradarCam(Camera):
     """ Deadline for image refresh """
     _deadline = None        # type: Optional[datetime]
     _name = ""              # type: str
-    _delta = 0.0         # type: float
+    _delta = 600.0          # type: float
```

Comment:
I do not know how I can add type annotations while supporting Python 3.5 (without [PEP 526](https://www.python.org/dev/peps/pep-0526/)) support otherwise.

```py
class Foo(object):
  bar  # type: int
```
Gives a `NameError: name 'bar' is not defined`

What would you propose?

Reply:
It should work in init method.

Reply:
Does this work? (I'm on mobile at the moment)
```py
class Foo:

    def __init__(self):
        self.bar = None  # type: int
```

Reply:
"Kind of works": mypy detects the types and has editor support.

The `self.bar = 0  # type: int` style does not create `module.__annotations__`. Nor does `self.bar: int = 0`. But functionally they work: mypy gives errors if you assign a different type.

mypy also infers types, `self.bar = 2` becomes an int, `self.bar = untyped_dict['a']` does not and needs the annotation.

Since we are on Python 3.5 for now I think the goal is tool support thus moving to types in the constructor is a good alternative.

With Python 3.6+ the assignment is no longer needed and the annotations become viable. I wish the annotations were just stripped on build instead of having to use the legacy style, I prefer the explicit annotations by now.